test/lit/forward-declared-types.wast

```diff
@@ -1,17 +1,16 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
 ;; Test that types can be used before they are defined
 
 ;; RUN: wasm-opt %s -all -S -o - | filecheck %s
 
-;; CHECK: (type $func (func))
-;; CHECK: (type $none_=>_ref?|$struct| (func (result (ref null $struct))))
-;; CHECK: (type $struct (struct (field (ref $array)) (field (ref null $func))))
-;; CHECK: (type $array (array (rtt 2 $func)))
-
 (module
+  ;; CHECK:      (type $func (func))
+  ;; CHECK:      (type $struct (struct (field (ref $array)) (field (ref null $func))))
```

Comment:
This reordering is definitely confusing at first, but seems unavoidable for the reasons you mentioned. Ideally filecheck could have an "ignore order" mode, but I guess this way we also check the order does not change.

Reply:
Full-fledged FileCheck in LLVM actually does support ignoring order, but I'm not sure the python package does.